cmd/crio/main.go

```diff
@@ -269,6 +269,18 @@ func main() {
 			return err
 		}
 
+		// Print the current cli flags
+		flagNames := c.FlagNames()
+		logrus.Infof("Cli Flags used:")
+		for _, flagName := range flagNames {
+			flagvalues := c.Value(flagName)
```

Comment:
The CLI package supports many different types of flags: string, boolean, numeric, etc., and it also handles repeating flags, so flags where you can set multiple values. This is then exposed as an internal type backed by a slice. What does this mean in practice? An example:

- Simple flag, boolean type

```console
$ crio --enable-criu-support 
(...)
INFO[2024-05-09 06:16:57.114618031Z] Cli Flags used:                              
INFO[2024-05-09 06:16:57.114626859Z] Flag: enable-criu-support: true 
```

- Repeated (multi-value) flag, string type

```console
$ crio --metrics-collectors=test1 --metrics-collectors=test2
(...)
INFO[2024-05-09 06:16:57.114618031Z] Cli Flags used:            
INFO[2024-05-09 06:18:59.071785731Z] Flag: metrics-collectors: {[test1 test2] { false false} true false}  
```

This is somewhat because of how `%v` works. Also, we aren't getting a simple string type back, and this particular type does not seem to have a sensible `String()` function (to fulfil the `Stringer` interface) implemented. We need to turn:

`Flag: metrics-collectors: {[test1 test2] { false false} true false}`

into:

`Flag: metrics-collectors: [test1,test2]`

Which would be as clean as it gets for such multi-value flags.

Reply:
With the code changes I see the flags printed like this:

'''
INFO[2024-05-10 12:29:44.853966920+02:00] FLAG: --enable-criu-support="true"           
INFO[2024-05-10 12:29:44.854005569+02:00] FLAG: --metrics-collectors="[test1,test2]"   
'''

Are the quotes around the flag value ok or better without?

Reply:
That matches the output of some of the other projects from the Kubernetes ecosystem (like kubelet, for example). However, we don't need to strictly follow it if we feel it's not what we want.

@cri-o/cri-o-maintainers, thoughts? Yours too, @LenkaSeg :smile:

Reply:
I think it's beautiful :)

Reply:
I don't have a strong preference, quotes work fine for me